Sources/Rendering/OpenGL/glsl/vtkVolumeFS2.glsl

```diff
@@ -163,18 +163,22 @@ void main()
       dot(endVC, vPlaneNormal0),
       dot(endVC, vPlaneNormal2),
       dot(endVC, vPlaneNormal4));
+
+    // start slightly inside and apply some jitter
+    float jitter = texture2D(jtexture, gl_FragCoord.xy/32.0).r;
     vec3 vdelta = endvpos - vpos;
+    vpos = vpos + normalize(vdelta)*(0.01 + 0.98*jitter)*sampleDistance;
+
+    // update vdelta post jitter
+    vdelta = endvpos - vpos;
     float numSteps = length(vdelta) / sampleDistance;
     vdelta = vdelta / float(numSteps);
 
-    // start slightly inside and apply some jitter
-    float jitter = texture2D(jtexture, gl_FragCoord.xy/32.0).r;
-    vpos = vpos + vdelta*(0.01 + 0.98*jitter);
     vec4 color = vec4(0.0, 0.0, 0.0, 0.0);
 
     vec3 ijk = vpos * vVCToIJK;
     vdelta = vdelta * vVCToIJK;
-    int count = int(numSteps - 0.2); // end slightly inside
+    int count = int(numSteps - 0.05); // end slightly inside
```

Comment:
@martinken perhaps I am missing a lot of details here but why not check on opacity saturation or boundary of the box for termination? Is that because we want fix number of steps?